libs/checkpoint-postgres/langgraph/checkpoint/postgres/aio.py

```diff
@@ -460,9 +501,27 @@ def put_writes(
             writes (Sequence[Tuple[str, Any]]): List of writes to store, each as (channel, value) pair.
             task_id (str): Identifier for the task creating the writes.
         """
-        return asyncio.run_coroutine_threadsafe(
-            self.aput_writes(config, writes, task_id), self.loop
-        ).result()
+        return _run_sync(
+            self.aput_writes(config, writes, task_id),
+            self.loop,
+        )
+
+
+T = TypeVar("T")
+
+
+def _run_sync(coro: Awaitable[T], loop: asyncio.AbstractEventLoop) -> T:
+    try:
+        if asyncio.get_running_loop() is loop:
+            # Using `run_coroutine_threadsafe` will result in a deadlock.
+            # We first make the loop reentrant and then run the coroutine.
+            nest_asyncio.apply(loop)
```

Comment:
we cannot use nest_asyncio in a library, that is the sort of thing which can only be used in an application